### DISCUSSION TITLE
How do I get back 'Another Quick Switcher: Normal search'?

### DISCUSSION THREAD

**Discussion by claremacrae** [tadashi-aikawa/obsidian-another-quick-switcher#99] [Q&A] [answered]
Has `Another Quick Switcher: Normal search`  been renamed or removed?

I found this to be really useful - it did just what I wanted. But for some reason, it is no longer available in my setup.

### Expected behaviour

- Hitting `Cmd + O` would present an `Another Quick Switcher` file dialog
- Opening the Command Palette, and filtering on Another, would show `Another Quick Switcher: Normal search`

### Actual behaviour

- Hitting `Cmd + O` briefly inverts the colours of the File at the top of the screen - no file dialog appears
- Opening the Command Palette, there is no "Normal Search"

![image](https://user-images.githubusercontent.com/4840096/187508927-c39a794d-fd80-496f-a8dc-87f44e64049b.png)

### Background

At some point I assigned Cmd+O to `'Another Quick Switcher: Normal search'`, and added a sidebar icon for it:

![image](https://user-images.githubusercontent.com/4840096/187505425-3f2102ca-4632-4665-a383-474d50de2d0d.png)

That hotkey is still in my hotkey file - and those are the only two occurrences of `"O"` in the file:

![image](https://user-images.githubusercontent.com/4840096/187506251-c914e9da-8053-40c1-81aa-770f9ee6b183.png)

### Versions

- Obsidian 0.15.9
- Another Quick Switcher: 6.2.0 (same behaviour in "6.1.0")
- OS macOS 10.15.7
- Plugins enabled during this testing:

```json
[
  "customizable-sidebar",
  "obsidian-another-quick-switcher"
]
```

**Comment by claremacrae** [comment]
Going back through recent git history of my install for this plugin, reverting to its version `"5.4.1"`, the command `Another Quick Switcher: Normal search` exists.

**Reply by claremacrae** [reply]
Going back through the release history, 5.4.1 was the last of the 5.x series, so I'll stick with thank for now, as a workaround...

And try to remember not to update this plugin for now, until I learn the proper fix to get the command reinstated in 6.x..

**Comment by claremacrae** [comment]
I was going to paste in my settings file for this plugin, but there is none. So I must never have modified the settings.

**Comment by tadashi-aikawa** [comment, marked as answer]
> Has Another Quick Switcher: Normal search been renamed or removed?

@claremacrae 
Yes. "Normal search" has been removed.

Please use "Recommended search" (similar to "Normal search") or your customized command.

**Reply by tadashi-aikawa** [reply]
If you have what doesn't work as expected, please tell me the difference with "Normal Search", and I'll answer.

**Reply by claremacrae** [reply]
Thank you. What does Recommended Search do exactly please? How does it differ from Normal Search?

**Reply by claremacrae** [reply]
May I also suggest someone goes back and edits the release notes for 6.0 to state that Normal Search was removed? Plus any others that were removed. 

I hope it might save others the time I spent Investigating and trying to write a clear report.

**Reply by tadashi-aikawa** [reply]
> How does it differ from Normal Search?

There are bit differences in sort priorities.

`Recommended search`
```
Perfect word match
Prefix name match
Name match
Length
Tag match
Header match
Link match
Star
Last opened
Last modified
```

`Normal search`
```
Perfect word match
Prefix name match
Name match
Length
Last opened
Last modified
```

If you remove some priority items as the following, it will become "Normal search".

![image](https://user-images.githubusercontent.com/9500018/187568328-b708c5e9-5f5f-430c-840a-270108324551.png)

**Reply by tadashi-aikawa** [reply]
Ah... as you said, there are not about search commands. I'll add it 👍 

![image](https://user-images.githubusercontent.com/9500018/187569891-915dde81-3dd0-44f6-9a89-228357743081.png)

**Reply by claremacrae** [reply]
Thank you for the explanation of what Normal Search did. I can reproduce that.

Thank you for offering to update the release notes.

I should probably have also suggested update this location on the README to state what commands are removed 
https://github.com/tadashi-aikawa/obsidian-another-quick-switcher#for-users-who-use-earlier-than-v600

Many thanks for the great plugin, and for the help!

**Reply by tadashi-aikawa** [reply]
Updated README 👍 

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/blob/0aed662e21925c46bfeb6aa73fd823204ccb841a/README.md#for-users-who-use-earlier-than-v600

**Comment by claremacrae** [comment]
I was able to add the new Normal search, by editing the `data.json`, making a duplicate of "Recommended search", changing its name to "Normal search" and deleting the 4 highlighted sort fields.

However, doing this didn't fix my broken buttons in things like the custom side-bar plugin, and the hotkey.

For info, for anyone else doing the same update, It seems that the pattern for command names has changed:

```diff
- "obsidian-another-quick-switcher:normal-search"
+ "obsidian-another-quick-switcher:search-command_Normal search"
```

**Reply by tadashi-aikawa** [reply]
@claremacrae 

> "obsidian-another-quick-switcher:normal-search"

This hotkey setting is created by less than v6.0.0.

> "obsidian-another-quick-switcher:search-command_Normal search"

This hotkey setting is created by v6.0.0 or over. It should be fixed if you recreate settings about the custom side-bar plugin, and the hotkey.

By the way, v7.0.0 also has breaking changes related to the command-id generation. So I think it would be better after release (and installing) v7.0.0. (Please see #97)

**Reply by claremacrae** [reply]
Hi, thanks for the reply. Yes of course I understood that the change in command id was from the update to 6.0.

It just took a little while to work out the pattern and find the locations in my vault to update, so I wanted to share the info here for others.

It would really help if the update notes could in future be more detailed and specific about the nature of breaking changes.

**Reply by tadashi-aikawa** [reply]
Thank you for your explanation :)

I have just added a sentence to README 👍 

![image](https://user-images.githubusercontent.com/9500018/191024971-be957250-9412-44c5-9571-84c75dca6c0c.png)

**Reply by claremacrae** [reply]
Hi, I appreciate the thought, but would have no idea what to do with the information in that sentence.
Sorry.
